### PR TITLE
Added section on ByAll

### PR DESCRIPTION
**Pull request by shbenzer**
### **User description**
Added a section on ByAll to Locators

### Description
Added section ByAll to Locators
Added java example to locators.java
updated ByChained examples based on new example code lines
propagated changes to all translations

### Motivation and Context
increase comprehensiveness of documentation

### Types of changes

- [x] Change to the site (I have double-checked the Netlify deployment, and my changes look good)
- [x] Code example added (and I also added the example to all translated languages)
- [ ] Improved translation
- [ ] Added new translation (and I also added a notice to each document missing translation)

### Checklist

- [x] I have read the [**contributing**](https://selenium.dev/documentation/en/contributing/) document.
- [x] I have used [hugo](https://gohugo.io/) to render the site/docs locally and I am sure it works.

___

### **PR Type**
Documentation, Enhancement

___

### **Description**
- Added a new section on the `ByAll` class to the locators documentation, explaining its usage and providing examples.
- Updated the `ByChained` examples to reflect new code lines and improve clarity.
- Enhanced the Java test suite by adding a new test method `ByAllTest` to demonstrate the usage of the `ByAll` locator.
- Added a new dependency for the Scala library in the Java project's `pom.xml`.
- Propagated documentation changes to multiple translations including Japanese, Portuguese, and Chinese.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Dependencies</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>pom.xml</strong><dd><code>Add Scala library dependency to Java project</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

examples/java/pom.xml

- Added a new dependency for Scala library version 2.13.0.

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1939/files#diff-57fef07b0713fb4af5de90aa728eeb03d0497763a3bc9ffe1d551f85b88271b4">+6/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>LocatorsTest.java</strong><dd><code>Add ByAll locator test example in Java</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

examples/java/src/test/java/dev/selenium/elements/LocatorsTest.java

<li>Imported <code>ByAll</code> class.<br> <li> Added <code>ByAllTest</code> method to demonstrate usage of <code>ByAll</code>.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1939/files#diff-fb0196c82ef66308dc7f93c93b1552cc18a0c83380d38032b5814a69987ea4ec">+19/-0</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Documentation</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>locators.en.md</strong><dd><code>Document ByAll class and update ByChained examples</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/elements/locators.en.md

<li>Added documentation for <code>ByAll</code> class.<br> <li> Updated <code>ByChained</code> example references.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1939/files#diff-bad5c08d78f84e8bd7528d1a2a3181973375147d3f8bc5b9678cfce7dbe76aab">+28/-4</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>locators.ja.md</strong><dd><code>Document ByAll class and update ByChained examples in Japanese</code></dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/elements/locators.ja.md

<li>Added documentation for <code>ByAll</code> class.<br> <li> Updated <code>ByChained</code> example references.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1939/files#diff-06ad92a757672227e965b2d837608b89da6bb02315528815d9e6fa2eac9114af">+27/-3</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>locators.pt-br.md</strong><dd><code>Document ByAll class and update ByChained examples in Portuguese</code></dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/elements/locators.pt-br.md

<li>Added documentation for <code>ByAll</code> class.<br> <li> Updated <code>ByChained</code> example references.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1939/files#diff-d1f5d96e2cf4217d2473ebd28f060129cedf8bde199fc884493ffd01d205d994">+27/-3</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>locators.zh-cn.md</strong><dd><code>Document ByAll class and update ByChained examples in Chinese</code></dd></summary>
<hr>

website_and_docs/content/documentation/webdriver/elements/locators.zh-cn.md

<li>Added documentation for <code>ByAll</code> class.<br> <li> Updated <code>ByChained</code> example references.<br>

</details>

  </td>
  <td><a href="https://github.com/SeleniumHQ/seleniumhq.github.io/pull/1939/files#diff-275209ad35c57329488099cdbd775ba45e6386d21561836642e99c0b67721f10">+27/-3</a>&nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: examples/java/pom.xml [hunk 1]
```diff
@@ -48,6 +48,12 @@
             <artifactId>selenium-logger</artifactId>
             <version>2.4.0</version>
         </dependency>
+    
+      <dependency>
+        <groupId>org.scala-lang</groupId>
```
Comment (harsha509) [inline examples/java/pom.xml]: is this dep required ?

File: website_and_docs/content/documentation/webdriver/elements/locators.en.md [hunk 2]
```diff
@@ -370,12 +370,36 @@ others it's as simple as setting a parameter in the FindElement function
 
 ### ByChained
 
-The `ByChained` class enables you to _chain_ two By locators together. For example, instead of having to locate a parent element, and then a child element of that parent, you can instead combine those two `FindElement` functions into one.
+The `ByChained` class enables you to _chain_ two By locators together. For example, instead of having to locate a parent element, and then a child element of that parent, you can instead combine those two `FindElement()` functions into one.
```
Comment (harsha509) [inline website_and_docs/content/documentation/webdriver/elements/locators.en.md]: HI @shbenzer ,
Could you please limit the line length to 100 characters as per the style guidelines

See Line length in https://www.selenium.dev/documentation/about/style/

**Review by harsha509 — CHANGES_REQUESTED** [review]

File: website_and_docs/content/documentation/webdriver/elements/locators.en.md (hunk 2 above)
Comment (shbenzer) [reply website_and_docs/content/documentation/webdriver/elements/locators.en.md]: Ah, thought I fixed all of those but you're right that line is over 100. My bad

File: examples/java/pom.xml (hunk 1 above)
Comment (shbenzer) [reply examples/java/pom.xml]: I'll go through the dependencies and see if an array/list library is already being used in Java

Comment (shbenzer) [reply examples/java/pom.xml]: No it's not - it was erroneously added by vscode. I'm using java.util.List.

I'll remove it

Comment (harsha509) [reply examples/java/pom.xml]: Thank you!

File: website_and_docs/content/documentation/webdriver/elements/locators.en.md (hunk 2 above)
Comment (harsha509) [reply website_and_docs/content/documentation/webdriver/elements/locators.en.md]: Thank you!

**Review by harsha509 — APPROVED** [review]
Thank you @shbenzer !

**Pull request merged**